infra/pipeline.ts

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env node
+import codebuild = require('@aws-cdk/aws-codebuild');
+import codepipeline = require('@aws-cdk/aws-codepipeline');
+import actions = require('@aws-cdk/aws-codepipeline-actions');
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/core');
+
+/**
+ * Simple three-stage pipeline to build the base image for the local container endpoints image.
+ * [GitHub source] -> [CodeBuild build, pushes image to DockerHub] -> [CodeBuild Verify, verifies the pushed images]
+ *
+ * TODO: use Docker manifest to create manifest for images under a common tag and ECR public
+ */
+class EcsLocalContainerEndpointsImagePipeline extends cdk.Stack {
+  constructor(parent: cdk.App, name: string, props?: cdk.StackProps) {
+    super(parent, name, props);
+
+    // Instantiate pipeline
+    const pipeline = new codepipeline.Pipeline(this, 'Pipeline', {
+      pipelineName: 'local-container-endpoints-image',
+    });
+
+    // Source stage
+    // Secret under ecs-local-container-endpoints+release@amazon.com
+    const githubAccessToken = cdk.SecretValue.secretsManager('EcsDevXGitHubToken');
+
+    const sourceOutput = new codepipeline.Artifact('SourceArtifact');
+    const sourceAction = new actions.GitHubSourceAction({
+      actionName: 'GitHubSource',
+      owner: 'awslabs',
+      repo: 'amazon-ecs-local-container-endpoints',
+      oauthToken: githubAccessToken,
+      branch: 'mainline',
+      output: sourceOutput
+    });
+
+    pipeline.addStage({
+      stageName: 'Source',
+      actions: [sourceAction],
+    });
+
+    // Build stage containing build project for each architecture image
+    const buildStage = pipeline.addStage({
+      stageName: 'Build',
+    });
+
+    const verifyStage = pipeline.addStage({
+      stageName: 'Verify',
+    });
+
+    const platforms = [
+      {'arch': 'amd64', 'buildImage': codebuild.LinuxBuildImage.AMAZON_LINUX_2_3},
+      {'arch': 'arm64', 'buildImage': codebuild.LinuxBuildImage.AMAZON_LINUX_2_ARM},
+    ];
+
+    // Create build and verify project for each platform
+    for (const platform of platforms) {
+      const arch = platform['arch'];
+
+      const buildProject = new codebuild.PipelineProject(this, `BuildImage-${arch}`, {
+        buildSpec: codebuild.BuildSpec.fromSourceFilename('./buildspec.yml'),
+        environment: {
+          buildImage: platform['buildImage'],
+          privileged: true,
+          environmentVariables: {
+            ARCH_SUFFIX: { value: arch },
+          }
+        }
+      });
+
+      const verifyProject = new codebuild.PipelineProject(this, `VerifyImage-${arch}`, {
+        buildSpec: codebuild.BuildSpec.fromSourceFilename('./buildspec_verify.yml'),
+        environment: {
+          buildImage: platform['buildImage'],
+          privileged: true,
+          environmentVariables: {
+            ARCH_SUFFIX: { value: arch },
+          }
+        }
+      });
+
+      buildProject.addToRolePolicy(new iam.PolicyStatement({
```

Comment:
Do we need ECR permissions? I might be misunderstanding it but it looks like we are only pushing and pulling from Dockerhub so I wonder if we can delete those permissions 🤔

Reply:
We don't currently, but I kept them for when we eventually include ECR repos into our pipeline, which was a suggestion I received from @samuelkarp.

Reply:
See: 20f2349b480a30872863370a450be978ae987c8c